packages/stark-ui/src/modules/date-range-picker/components/date-range-picker.component.ts

```diff
@@ -117,67 +191,140 @@ export class StarkDateRangePickerComponent extends AbstractStarkUiComponent impl
 	public dateRangeChanged: EventEmitter<StarkDateRangePickerEvent> = new EventEmitter<StarkDateRangePickerEvent>();
 
 	/**
-	 * Reference to the start datepicker embedded in this component
+	 * Filter function or a string
+	 * Will be applied to both date-picker
 	 */
-	@ViewChild("startPicker")
-	public startPicker!: StarkDatePickerComponent;
+	@Input()
+	public dateFilter?: StarkDatePickerFilter;
 
 	/**
-	 * Reference to the end datepicker embedded in this component
+	 * Timestamp Mask Configuration to apply on the start/end date-picker.
+	 * If `true` is passed, the default mask config is applied: {DEFAULT_DATE_MASK_CONFIG|DEFAULT_DATE_MASK_CONFIG}
+	 * If `false` is passed or if `dateMask` is not present, the directive is disabled.
+	 * If a `StarkTimestampMaskConfig` is passed, it is set as the date mask config.
 	 */
-	@ViewChild("endPicker")
-	public endPicker!: StarkDatePickerComponent;
+	@Input()
+	public dateMask?: StarkDatePickerMaskConfig;
+
+	/**
+	 * Whether the datepickers are disabled
+	 */
+	@Input()
+	public set disabled(val: boolean) {
+		if (val) {
+			this.startDateFormControl.disable();
+			this.endDateFormControl.disable();
+		} else {
+			this.startDateFormControl.enable();
+			this.endDateFormControl.enable();
+		}
+	}
 
 	/**
-	 * Class constructor
-	 * @param logger - The logger of the application
-	 * @param renderer - Angular Renderer wrapper for DOM manipulations.
-	 * @param elementRef - Reference to the DOM element where this directive is applied to.
+	 * HTML "name" attribute of the element.
 	 */
+	@Input()
+	public rangePickerId = "";
+
+	/**
+	 * HTML "name" attribute of the element.
+	 */
+	@Input()
+	public rangePickerName = "";
+
 	public constructor(
 		@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService,
+		private injector: Injector,
 		protected renderer: Renderer2,
 		protected elementRef: ElementRef
 	) {
 		super(renderer, elementRef);
+		this.subs.push(
+			this.startDateFormControl.valueChanges.subscribe(() => this.onDateChanged()),
+			this.endDateFormControl.valueChanges.subscribe(() => this.onDateChanged())
+		);
 	}
 
 	/**
-	 * Component lifecycle hook
+	 * Angular lifecycle method
 	 */
 	public ngOnInit(): void {
+		this._setupNgControl();
 		this.logger.debug(componentName + ": component initialized");
 	}
 
 	/**
-	 * Handle the date changed on the start datepicker
+	 * Link the passed control (if one is available) to the internal formControls.
+	 * @internal
+	 * @ignore
 	 */
-	public onDateStartChanged(date: Date): void {
-		this.startDate = date;
-		this.checkDates();
+	private _setupNgControl(): void {
+		// Get the ngControl from the injector
+		try {
+			const ngControl = this.injector.get<NgControl>(<Type<NgControl>>NgControl);
```

Comment:
Here, you can simplify the code to avoid having a `try...catch` block. Something like this:

```typescript
const ngControl = this.injector.get<NgControl>(<Type<NgControl>>NgControl, <any>null);  // default value: null

if (ngControl !== null) {
    ngControl.valueAccessor = this;
}
```